{
    "body": "Fixes #4919\r\n\r\nFor reasons that I do not entire understand, duplicate_stream() throws\r\nan '[Errno 22] Invalid argument' exception when it tries to\r\n`os.fdopen()` the duplicated file descriptor generated by\r\n`os.dup(original.fileno())`.  See spack/llnl/util/lang.py, line\r\n394-ish.\r\n\r\nThis happens when run under `nohup`, which supposedly has hooked\r\n`stdin` to `/dev/null`.\r\n\r\nIt seems like opening and using `devnull` on the `input_stream` in\r\nthis situation is a reasonable way to handle the problem.",
    "user": "hartzell",
    "url": "https://api.github.com/repos/spack/spack/issues/4926",
    "updated_at": "2017-08-03 17:33:27",
    "created_at": "2017-07-29 20:19:32",
    "closed_at": "2017-08-03 17:33:16",
    "state": "closed",
    "title": "Fix crashes when running spack install under nohup",
    "number": 4926,
    "milestone": null,
    "labels": [],
    "id": 246543104,
    "html_url": "https://github.com/spack/spack/pull/4926",
    "assignees": [],
    "comments": 3
}